### DISCUSSION TITLE
argocd cluster add - Using a Service Account or using a K88s secret

### DISCUSSION THREAD

**Discussion by Josephred999** [argoproj/argo-cd#15294] [Q&A]
Hello,
When adding clusters to the argocd, currently I am leveraging my local kubeconfig (😬), which is not a better solution, and I am still in a POC mode. 

Before I put it in my Production, I need to migrate this to either a service account or vault or a proper method which I am not really familiar with.

Appreciate any tips here folks.

**Comment by FernandoMiguel** [comment]
@Josephred999 where are your clusters located?
for AWS EKS it's super easy. you just write the authconfig into git and let argo deploy that

**Reply by Josephred999** [reply]
Nope, these are my local K8s clusters. Not in cloud.

**Reply by FernandoMiguel** [reply]
@Josephred999 an approach we used before was to store those secrets in vault and have ESO pull them, and a custom app split the secret, and create individual secrets per cluster
